### DISCUSSION TITLE
Cleanlab on regression problems?

### DISCUSSION THREAD

**Discussion by mengqvist** [cleanlab/cleanlab#137] [Q&A] [answered]
Hi! 

I was wondering whether there's a natural way to extend cleanlab to identify label noise on regression problems, as opposed to classification problems? I am aware of some other frameworks, such as Shapley values, but it would be great if I could use cleanlab directly. I could bin the data, but that's not particularly elegant.

Thanks!

**Comment by mengqvist** [comment, marked as answer]
Ah! After poking about I found a [note](https://github.com/cleanlab/cleanlab/blob/master/cleanlab/version.py) that extension to regression problems is planned as a future feature. Is this actively worked on?

**Reply by jwmueller** [reply]
Yes we just beginning to actively think about adding this extension to regression tasks!

@mengqvist Would you be interested in helping contribute on this effort?
If so, we'd love to hear your insights and learn more about your use-case!  

If you prefer more seamless communication, you can message us in our Slack channel:
https://join.slack.com/t/cleanlab-community/shared_invite/zt-17lszn4hv-gg2FhZPXYfljq_l01uo92g

**Reply by mglowacki100** [reply]
@jwmueller Are you still working on regression for cleanlab?
Have you consider following idea:
1. Let's start with 'ordinal regression', e.g. we rank movies with one, two or three stars. It could be relabelled to 0, 0.5, 1 and treated as ordinary regression. Next most of regression predictions would be between 0 and 1; predictions above 1 could be capped to 1,  analogously predictions below 0 could be treated as 0. Those predictions could be treated as `pred_probs` for 'DataLab.find_issues(...)`. Obviously, this could be generalized to any number of "stars".
3. If idea from point **1** works, then general case of regression could be transformed to it by:
- scaling to [0,1]
- bucketing
with some nitty-gritty details like outliers treatment and optimal number of buckets

**Reply by jwmueller** [reply]
@mglowacki100 Support for regression datasets is already available in the package today, just not in the stable release yet:  https://docs.cleanlab.ai/master/tutorials/regression.html 

To use it, simply do:  `pip install git+https://github.com/cleanlab/cleanlab.git`

You can read about the method here:  [Detecting Errors in Numerical Data via any Regression Model
](https://arxiv.org/abs/2305.16583)

**Reply by mglowacki100** [reply]
@jwmueller Thanks for prompt reply! Do you plan to extend `DataLab` for regression?

**Reply by jwmueller** [reply]
Yes eventually, but it's not too high on the current priority list.

If you'd like to help with that extension, we could certainly use assistance!

**Reply by mglowacki100** [reply]
@jwmueller I've created separate issue for `DataLab` for regression task with some questions/initial investigations https://github.com/cleanlab/cleanlab/issues/765.
